plugins/inventory/now.py

```diff
@@ -454,12 +454,27 @@ def _merge_instance_config(self, instance_config, instance_env):
         return instance
 
     def _get_instance_from_env(self):
+        def get_secret_from_env():
+            for arg in ("SN_CLIENT_SECRET", "SN_SECRET_ID"):
+                value = os.getenv(arg)
+                if value is not None:
+                    if arg == "SN_SECRET_ID":
+                        # Remove this in 3.0.0
+                        self.display.deprecated(
+                            "Setting environment variable 'SN_SECRET_ID' is being removed "
+                            "in favor of 'SN_CLIENT_SECRET'",
+                            version="3.0.0",
+                            collection_name="servicenow.itsm",
+                        )
+                    return value
+            return None
```

Comment:
I think that this is now better in terms that we encourage the use of the new variable, while preserving backwards compatibility. However, if both variables are defined (unintentionally), it might be confusing to a user from where the value is taken from. I would be more explicit about it.

Reply:
If both values are set then `SN_CLIENT_SECRET` will always be set.